lib/reverse_sentence.rb

```diff
@@ -1,6 +1,36 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n) because method reverse_range is O(n) due to the single times loop through the collection, 
+# and we call reverse_range as many times as needed but the calls aren't nested and the number of calls made 
+# is independent of the size of the collection
+# Space complexity: O(n), no new structures created or maintained
 def reverse_sentence(my_sentence)
```

Comment:
Outstanding submission.  Well done.

That said space complexity is measured by the additional memory used based on the input, and your space complexity is O(1) because you don't create any new structures.